### PR TITLE
disable warnings summary for pytest

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

There are literally thousands of lines of output from pytest informing us about warnings. 

```
tests/checkpointing/test_model_checkpoint.py::test_model_checkpoint_score_and_ckpt_val_check_interval[0.42-False-False]
  /__w/pytorch-lightning/pytorch-lightning/pytorch_lightning/callbacks/model_checkpoint.py:622: UserWarning: Checkpoint directory /tmp/pytest-of-root/pytest-0/test_model_checkpoint_score_an8 exists and is not empty.
    rank_zero_warn(f"Checkpoint directory {dirpath} exists and is not empty.")

```
It's simply not practical. 
This PR disables this nonsense of endless scrolling.

The statistic of how many warnings there are in total is still visible like so:
`==== 44 passed, 7 skipped, 217 warnings in 5.33s ====
`
## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Comment by ananthsub** [comment]
> There are literally thousands of lines of output from pytest informing us about warnings.

how much of the warnings would you estimate is due to overzealous warnings by the framework vs tests having real issues?
if we turn off warnings like this, would users turn off warnings too?

**Comment by awaelchli** [comment]
@ananthsub I would say the majority is definitely from the framework. The majority of warnings are "friendly" messages for users who haven't fully configured and tuned the settings. But in our tests, messages like these do not matter. For example, since we use BoringModel in practically every test, it will always raise this warning:

> The dataloader, train_dataloader, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument`

**Review by SeanNaren — APPROVED** [review]

File: setup.cfg [hunk 1]
```diff
@@ -24,6 +24,7 @@ addopts =
     --strict
     --doctest-modules
     --color=yes
+    --disable-pytest-warnings
```
Comment (awaelchli) [inline setup.cfg:27]:
```suggestion
    --disable-pytest-warnings

```

**Review by ananthsub — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGMT !
